Rename HighlightData to RenderHighlight and SelectionRangeData to RenderSelection as part of rename work to clarify classes around highlights.

megangardner opened the pull request:
#### ab192be42c5ef4cf2771f3ade70eba6a4caab3bc
<pre>
Rename HighlightData to RenderHighlight and SelectionRangeData to RenderSelection as part of rename work to clarify classes around highlights.
<a href="https://bugs.webkit.org/show_bug.cgi?id=261634">https://bugs.webkit.org/show_bug.cgi?id=261634</a>
rdar://115589022

Reviewed by Richard Robinson.

Several classes associated with Highlights have a very confusing naming
structure. This is the third rename of several to help clarify
what these classes are and their relationship to each other.

* Source/WebCore/rendering/RenderHighlight.cpp: Renamed from Source/WebCore/rendering/HighlightData.cpp.
(WebCore::RenderRangeIterator::RenderRangeIterator):
(WebCore::RenderRangeIterator::current const):
(WebCore::RenderRangeIterator::next):
(WebCore::RenderRangeIterator::checkForSpanner):
(WebCore::rendererAfterOffset):
(WebCore::RenderHighlight::setRenderRange):
(WebCore::RenderHighlight::highlightStateForRenderer):
(WebCore::RenderHighlight::highlightStateForTextBox):
(WebCore::RenderHighlight::rangeForTextBox):
* Source/WebCore/rendering/RenderHighlight.h: Renamed from Source/WebCore/rendering/HighlightData.h.
(WebCore::RenderRange::RenderRange):
(WebCore::RenderRange::start const):
(WebCore::RenderRange::end const):
(WebCore::RenderRange::startOffset const):
(WebCore::RenderRange::endOffset const):
(WebCore::RenderHighlight::RenderHighlight):
(WebCore::RenderHighlight::get const):
(WebCore::RenderHighlight::start const):
(WebCore::RenderHighlight::end const):
(WebCore::RenderHighlight::startOffset const):
(WebCore::RenderHighlight::endOffset const):
* Source/WebCore/rendering/RenderSelection.cpp: Renamed from Source/WebCore/rendering/SelectionRangeData.cpp.
(WebCore::rendererAfterOffset):
(WebCore::isValidRendererForSelection):
(WebCore::containingBlockBelowView):
(WebCore::collectSelectionData):
(WebCore::RenderSelection::RenderSelection):
(WebCore::RenderSelection::set):
(WebCore::RenderSelection::clear):
(WebCore::RenderSelection::repaint const):
(WebCore::RenderSelection::collectBounds const):
(WebCore::RenderSelection::apply):
* Source/WebCore/rendering/RenderSelection.h: Renamed from Source/WebCore/rendering/SelectionRangeData.h.
(WebCore::RenderSelection::bounds const):
(WebCore::RenderSelection::boundsClippedToVisibleContent const):

Canonical link: <a href="https://commits.webkit.org/268089@main">https://commits.webkit.org/268089@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ab8798950b4b5ba965a1fa28f97616fc7bf0b06

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18659 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19000 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19603 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20522 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17466 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18855 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22309 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19141 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19315 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18884 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19042 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16252 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21398 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16265 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17012 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23458 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/17290 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17184 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21351 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17780 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15091 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16833 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4431 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21200 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17619 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
LGTM. I think you need to add the new header files to the CMake source list

megangardner committed:
Rename HighlightData to RenderHighlight and SelectionRangeData to RenderSelection as part of rename work to clarify classes around highlights.
https://bugs.webkit.org/show_bug.cgi?id=261634
rdar://115589022

Reviewed by Richard Robinson.

Several classes associated with Highlights have a very confusing naming
structure. This is the third rename of several to help clarify
what these classes are and their relationship to each other.

* Source/WebCore/rendering/RenderHighlight.cpp: Renamed from Source/WebCore/rendering/HighlightData.cpp.
(WebCore::RenderRangeIterator::RenderRangeIterator):
(WebCore::RenderRangeIterator::current const):
(WebCore::RenderRangeIterator::next):
(WebCore::RenderRangeIterator::checkForSpanner):
(WebCore::rendererAfterOffset):
(WebCore::RenderHighlight::setRenderRange):
(WebCore::RenderHighlight::highlightStateForRenderer):
(WebCore::RenderHighlight::highlightStateForTextBox):
(WebCore::RenderHighlight::rangeForTextBox):
* Source/WebCore/rendering/RenderHighlight.h: Renamed from Source/WebCore/rendering/HighlightData.h.
(WebCore::RenderRange::RenderRange):
(WebCore::RenderRange::start const):
(WebCore::RenderRange::end const):
(WebCore::RenderRange::startOffset const):
(WebCore::RenderRange::endOffset const):
(WebCore::RenderHighlight::RenderHighlight):
(WebCore::RenderHighlight::get const):
(WebCore::RenderHighlight::start const):
(WebCore::RenderHighlight::end const):
(WebCore::RenderHighlight::startOffset const):
(WebCore::RenderHighlight::endOffset const):
* Source/WebCore/rendering/RenderSelection.cpp: Renamed from Source/WebCore/rendering/SelectionRangeData.cpp.
(WebCore::rendererAfterOffset):
(WebCore::isValidRendererForSelection):
(WebCore::containingBlockBelowView):
(WebCore::collectSelectionData):
(WebCore::RenderSelection::RenderSelection):
(WebCore::RenderSelection::set):
(WebCore::RenderSelection::clear):
(WebCore::RenderSelection::repaint const):
(WebCore::RenderSelection::collectBounds const):
(WebCore::RenderSelection::apply):
* Source/WebCore/rendering/RenderSelection.h: Renamed from Source/WebCore/rendering/SelectionRangeData.h.
(WebCore::RenderSelection::bounds const):
(WebCore::RenderSelection::boundsClippedToVisibleContent const):

Canonical link: https://commits.webkit.org/268089@main
